src/sort-scripts.ts

```diff
@@ -3,31 +3,57 @@ import orderBy from 'sort-order';
 import { PackageJson } from './types';
 
 const PRE_OR_POST_PREFIX = /^(pre|post)/;
+const NPM_RUN_ALL_SEPARATOR = /([:/])/;
+const NPM_BUILTIN_SCRIPTS = ['install', 'prepare', 'pack', 'publish', 'restart', 'start', 'stop', 'test', 'version', 'uninstall'];
+
+function parseScriptName(scripts: PackageJson['scripts'], script: string): { prefix: '' | 'pre' | 'post'; name: string } {
+  if (script === 'prepublishOnly') return { prefix: 'pre', name: 'publish' };
+  const prefixMatch = PRE_OR_POST_PREFIX.exec(script);
+  const prefix = prefixMatch ? prefixMatch[0] as 'pre' | 'post' : '';
+  const name = script.slice(prefix.length);
+  if (scripts!.hasOwnProperty(name) || NPM_BUILTIN_SCRIPTS.includes(name)) return { prefix, name };
+  return { prefix: '', name: script };
+}
 
 // Sort alphabetically by script name excluding pre/post prefixes
-function scriptName(...args: [string, string]): 1 | 0 | -1 {
-  const [a, b] = args.map((arg) => arg.replace(PRE_OR_POST_PREFIX, ''));
-  if (a !== b) {
-    return a < b ? -1 : 1;
-  } else {
+function scriptName(scripts: PackageJson['scripts']) {
+  return function (a: string, b: string): 1 | 0 | -1 {
+    const an = parseScriptName(scripts, a).name;
+    const bn = parseScriptName(scripts, b).name;
+    const aa = an.split(NPM_RUN_ALL_SEPARATOR);
```

Comment:
**Suggestion:** Perhaps this parsing should happen in `parseScriptName`

Perhaps parsing would creating something like:

```ts
interface ParsedScriptName {
  prefix?: 'pre' | 'post';

  /**
   * The string _preceding_ a separator such as `:` or `/`.
   * For example, for an input `prettier:lint`, the category would be `prettier`.
   */
  category?: string;

  /**
   * The string _after_ a separator such as `:` or `/`.
   * For example, for an input `prettier:lint`, the name would be `lint`.
   */
  name: string;

  /**
   * The full original name of the script
   */
  original: string;
};
```

Reply:
I think it will be difficult to separate only the category and name.
npm-run-all handles names with multiple separators, such as lint:eslint:fix. When handling these names, if we only split once, there will be problems with sorting in the second and subsequent blocks.
Therefore, I think it would be better to use split to divide the string.